typings/index.d.ts

```diff
@@ -1119,6 +1124,49 @@ export class GuildPreview extends Base {
   public toString(): string;
 }
 
+export class GuildScheduledEvent<S extends GuildScheduledEventStatus = GuildScheduledEventStatus> extends Base {
+  private constructor(client: Client, data: RawGuildScheduledEventData);
+  public id: Snowflake;
+  public guildId: Snowflake;
+  public channelId: Snowflake | null;
+  public creatorId: Snowflake | null;
+  public name: string;
+  public description: string | null;
+  public scheduledStartTime: number | null;
+  public scheduledEndTime: number | null;
+  public privacyLevel: GuildScheduledEventPrivacyLevel;
+  public status: S;
+  public entityType: GuildScheduledEventEntityType;
+  public entityId: Snowflake | null;
+  public entityMetadata: GuildScheduledEventEntityMetadata;
+  public userCount: number | null;
+  public creator: User | null;
+  public readonly createdTimestamp: number;
+  public readonly createdAt: Date;
+  public readonly scheduledStartAt: Date;
+  public readonly scheduledEndAt: Date | null;
+  public readonly channel: VoiceChannel | StageChannel | null;
+  public readonly guild: Guild | null;
+  public readonly url: string;
+  public createInviteURL(options?: CreateGuildScheduledEventInviteURLOptions): Promise<string>;
+  public edit(options: GuildScheduledEventEditOptions): Promise<GuildScheduledEvent>;
+  public delete(): Promise<GuildScheduledEvent>;
+  public setName(name: string): Promise<GuildScheduledEvent>;
+  public setScheduledStartTime(scheduledStartTime: DateResolvable): Promise<GuildScheduledEvent>;
+  public setScheduledEndTime(scheduledEndTime: DateResolvable | null): Promise<GuildScheduledEvent>;
+  public setDescription(description: string): Promise<GuildScheduledEvent>;
+  public setStatus(status: GuildScheduledEventSetStatusArg<S>): Promise<GuildScheduledEvent>;
+  public setLocation(location: string): Promise<GuildScheduledEvent>;
```

Comment:
You'll want to get the new status from edit/setStatus somehow and pass it down
```suggestion
  public edit(options: GuildScheduledEventEditOptions): Promise<GuildScheduledEvent>;
  public delete(): Promise<GuildScheduledEvent<S>>;
  public setName(name: string): Promise<GuildScheduledEvent<S>>;
  public setScheduledStartTime(scheduledStartTime: DateResolvable): Promise<GuildScheduledEvent<S>>;
  public setScheduledEndTime(scheduledEndTime: DateResolvable | null): Promise<GuildScheduledEvent<S>>;
  public setDescription(description: string): Promise<GuildScheduledEvent<S>>;
  public setStatus(status: GuildScheduledEventSetStatusArg<S>): Promise<GuildScheduledEvent>;
  public setLocation(location: string): Promise<GuildScheduledEvent<S>>;
```

Reply:
✔️